src/cloudant/error.py

```diff
@@ -61,18 +71,116 @@ class ResultException(CloudantException):
     """
 
     def __init__(self, code=100, *args):
-        messages = {
-            100:'A general result exception was raised.',
-            101:'Failed to interpret the argument {0} as a valid key value or as a valid slice.',
-            102:'Cannot use {0} when performing key access or key slicing.  Found {1}.',
-            103:'Cannot use {0} for iteration.  Found {1}.',
-            104:'Invalid page_size: {0}'
-        }
-
         try:
-            msg = messages[code].format(*args)
+            msg = RESULT[code].format(*args)
         # pylint: disable=broad-except
         except Exception:
             code = 100
-            msg = messages[code]
+            msg = RESULT[code]
         super(ResultException, self).__init__(msg, code)
+
+
+class CloudantClientException(CloudantException):
+    """
+    Provides a way to issue Cloudant library client specific exceptions.
+
+    :param int code: A code value used to identify the client exception.
+    :param args: A list of arguments used to format the exception message.
+    """
+    def __init__(self, code=None, *args):
+        try:
+            msg = CLIENT[code].format(*args)
+        # pylint: disable=broad-except
+        except Exception:
+            msg = CLIENT[code]
```

Comment:
We could have a default `100` error code that says something like `A general Cloudant exception was raised.` in each exception class.  I'm not sure how helpful this message is but I also wouldn't want another exception to happen on this line.